### DISCUSSION TITLE
Porting to Other Desktop Environments

### DISCUSSION THREAD

**Discussion by TrentSaggus** [material-shell/material-shell#685] [General]
Hey guys, and thanks for such an awesome shell.

I was wondering if there are any plans to bring this to any other DE's, specifically KDE?

There are some great features of KDE that I know I want to use, so there must also be other users out there too wishing the same, but I'd love to be able to use the brilliant shell on top of all of that.

Thanks again for the great work!

**Comment by PapyElGringo** [comment]
A Shell need to be very integrated in the environment and the work needed to maintain several environment is not doable only by me. It possible in the future that I remake all the Shell on Flutter and become platform independant.

**Reply by Alex-Bujorianu** [reply]
It would be great if Material Shell could become its own desktop environment. The reason is because, like many users, I am experiencing a lot of issues: regular freezes, invisible text in the search bar, and no text description for apps in the + menu. It’s very hard to figure out if the issue is Material Shell itself or some complex interaction between Material Shell and another extension. Realistically, gnome shell isn’t usable without extensions anyway. Furthermore, I am sure the fact that you have to maintain multiple versions for each gnome release causes a great maintenance burden.  

In my inexpert opinion, the best solution would be to take one version of Gnome + Material Shell and call it Material Desktop. You will have only one version to maintain and you’ll be able to fix more bugs/issues. Extension developers would only have to target one version. Improvements to Mutter could be patched in from upstream. (Though most of the issues stem from gnome-shell, right?)
